packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -58,14 +58,23 @@ import { RemoteDocumentChangeBuffer } from './remote_document_change_buffer';
 import { IterateOptions, SimpleDbStore } from './simple_db';
 import { ObjectMap } from '../util/obj_map';
 
-export class IndexedDbRemoteDocumentCache implements RemoteDocumentCache {
+export interface IndexedDbRemoteDocumentCache extends RemoteDocumentCache {
+  // The IndexedDbRemoteDocumentCache doesn't implement any methods on top
+  // of RemoteDocumentCache. This class exists for consistency.
+}
+
+/**
+ * The RemoteDocumentCache for IndexedDb. To construct, invoke
+ * `newIndexedDbRemoteDocumentCache()`.
+ */
+export class IndexedDbRemoteDocumentCacheImpl implements RemoteDocumentCache {
```

Comment:
Can "export" be removed from the declaration of `IndexedDbRemoteDocumentCacheImpl`? It doesn't appear to be used outside of this file and its name implies that it is an implementation detail that should not be exposed.

Reply:
The entire idea behind the XYZ and XYZImpl is that XYZImpl is not exported, so that the public fields in XYZImpl are only visible within the module. So, yes, this should not be exported.